### ISSUE TITLE
Settings page sometimes fails to access underlying settings model in background script

### ISSUE THREAD

**Issue by atomGit** [SmartRSS/Smart-RSS#141]
when i deleted all articles in the pinned group and checked the settings, the contents of the 'Auto pin articles matching' option were deleted as well

**Comment by zakius** [comment]
Can you try finding more precise reproduction steps? In the mentioned scenario it can't happen, there must be something else

**Comment by atomGit** [comment]
something very strange is happening and i don;t know how to reproduce

when 2.21 was released, i tried refreshing the feeds and the counter stuck at 0/2 - re-opening SRSS didn't work - only after restarting FF did a refresh work

so i entered a comma separated string in the filter option, as in 'italy,breaking.veritas' and refreshed the feeds - as expected, feeds matching my filters were linked in the 'pinned' group - i selected all of them (ctrl+a) and deleted them (del key), then looked opened the options again and the filter string was gone - blank

in trying to reproduce the issue, i located a new feed and entered a filter keyword in the options, then i added the feed expecting to find pinned items that i knew matched the filter word i used - instead, no items were pinned and again, the filter string was empty - so i think i closed and re-opened options and the **original** string was back 'italy,breaking.veritas'

i'll play with this some more and see if i can pin something down

**Comment by atomGit** [comment]
ps. give me some time to get dig into this further - got something i need to do

**Comment by zakius** [comment]
have you managed to find anything about it?

**Comment by atomGit** [comment]
there is an underlying problem - i was getting ready to test again, so i opened SRSS, then settings, then i deleted the existing filter/pin string and entered a new one, then closed the settings tab and immediately re-opened it from SRSS and the filter string was blank

next i closed SRSS and re-opened it -> settings and the filter string was an old one, not the last one i entered

**Comment by atomGit** [comment]
note that i tested with a default firefox (v84) profile and a fresh install of SRSS with default settings

**Comment by atomGit** [comment]
the title of this issue should probably be changed since the emptying of the filter/pin strings is not dependent on the deleting of pinned articles

**Comment by zakius** [comment]
It seems to be something else, closing for now, if issue surfaces again please open new issue
